packages/myst-frontmatter/src/kernelspec/validators.ts

```diff
@@ -6,29 +6,52 @@ import {
   validateObject,
   validateObjectKeys,
   validateString,
+  validationWarning,
 } from 'simple-validators';
 import type { KernelSpec } from './types.js';
 
-const KERNELSPEC_KEYS = ['name', 'language', 'display_name', 'argv', 'env'];
+const KERNELSPEC_KEYS = ['name', 'display_name', 'language', 'argv', 'env'];
 
 /**
  * Validate KernelSpec object
  */
 export function validateKernelSpec(input: any, opts: ValidationOptions) {
   const value = validateObjectKeys(input, { optional: KERNELSPEC_KEYS }, opts);
   if (value === undefined) return undefined;
-  const output: KernelSpec = {};
+
+  // name is a required member of KernelSpec, but hasn't previously been required
+  let name: string;
   if (defined(value.name)) {
-    output.name = validateString(value.name, incrementOptions('name', opts));
-  }
-  if (defined(value.language)) {
-    output.language = validateString(value.language, incrementOptions('language', opts));
+    const validatedName = validateString(value.name, incrementOptions('name', opts));
+    if (validatedName === undefined) return undefined;
```

Comment:
Is this correct? If it doesn't validate we bail and get no kernel?

Reply:
I think that makes sense.

Reply:
Yes, if the user gives a `name`, it must be valid. Otherwise, we handle the case that it's omitted for legacy reasons (up until now, it wasn't required for execution).